### ISSUE TITLE
Avoid List (or any other LinearSeq) for indexed accesses

### ISSUE THREAD

**Issue by mpasa** [nrinaudo/scala-best-practices#10]
A lot of people new to Scala tend to use `List` as the Java's interface, without knowing that is a Linked List. Maybe we can add a new section for data structures/collections?

**Comment by nrinaudo** [comment]
Interesting thought. I like the idea of a section on data structures and collections - and actually started off with one - but with the collection redesign looming close, I'm worried that it'll be obsolete before it's even released :)

If you have suggestions for other things that should go in the collections section, please post them! Enough of them (that I can confirm still apply in 2.13) and  I'll get to it.

**Comment by mpasa** [comment]
Yes, maybe it's better to wait for 2.13 with the new collections. RC1 is due by January 30, so I don't think we will have to wait a lot.
